perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java

```diff
@@ -481,23 +481,51 @@ protected void resolveMessage(String msg, Integer idOfMessage) throws InternalEr
 					//USER CERT DNS WILL BE SET (special method for updating)
 				} else if(this.attribute.getName().equals(cz.metacentrum.perun.core.api.AttributesManager.NS_USER_ATTR_VIRT + ":userCertDNs")) {
 					Map<String, String> certDNsMap = new HashMap<String, String>();
-					if(this.attribute.getValue() != null) certDNsMap = (Map) this.attribute.getValue();
+					if (this.attribute.getValue() != null) certDNsMap = (Map) this.attribute.getValue();
 					else certDNsMap = null;
 
-					if(certDNsMap == null || certDNsMap.isEmpty()) {
-						if(ldapConnector.userAttributeExist(this.user, "userCertificateSubject")) {
+					if (certDNsMap == null || certDNsMap.isEmpty()) {
+						if (ldapConnector.userAttributeExist(this.user, "userCertificateSubject")) {
 							updateUserAttribute("userCertificateSubject", null, LdapOperation.REMOVE_ATTRIBUTE, this.user);
 						}
 					} else {
-						Set<String> certSubjectsWithPrefixes =((Map) this.attribute.getValue()).keySet();
+						Set<String> certSubjectsWithPrefixes = ((Map) this.attribute.getValue()).keySet();
 						Set<String> certSubjectsWithoutPrefixes = new HashSet<>();
 						//remove prefixes from certificates
-						for(String key: certSubjectsWithPrefixes) {
+						for (String key : certSubjectsWithPrefixes) {
 							certSubjectsWithoutPrefixes.add(key.replaceFirst("^[0-9]+[:]", ""));
 						}
 						String[] subjectsArray = Arrays.copyOf(certSubjectsWithoutPrefixes.toArray(), certSubjectsWithoutPrefixes.toArray().length, String[].class);
 						ldapConnector.updateUsersCertSubjects(String.valueOf(this.user.getId()), subjectsArray);
 					}
+				//USER SCHAC HOME ORGANIZATIONS WILL BE SET (special method for updating)
+				} else if(this.attribute.getName().equals(cz.metacentrum.perun.core.api.AttributesManager.NS_USER_ATTR_VIRT + ":schacHomeOrganizations")) {
+					List<String> shacHomeOrganizationsList = new ArrayList<>();
+					if(this.attribute.getValue() != null) shacHomeOrganizationsList = (ArrayList) this.attribute.getValue();
```

Comment:
both if branches assign the same variable, a ternary operator used in variable initialization would be more concise, just one line instead of three